File: backend/ongil/src/main/java/kr/co/ongil/domain/user/dto/response/UserResponse.java
package kr.co.ongil.domain.user.dto.response;

import kr.co.ongil.domain.user.entity.User;
import lombok.Builder;
import lombok.Getter;

@Getter
@Builder
public class UserResponse {

    private Integer id;
    private String name;
    private String birth;
    private String phoneNumber;
    private String userType;
    private String profileImage;
    private String provider;

    public static UserResponse from(User user) {
        return UserResponse.builder()
                .id(user.getId())
                .name(user.getName())
                .birth(user.getBirth())
                .phoneNumber(user.getPhoneNumber())
                .userType(user.getUserType().name())
                .profileImage(user.getProfileImage())
                .provider(user.getProvider().name())
                .build();
    }
}

File: android/ongil/app/src/main/java/kr/co/ongil/data/model/user/UserResponse.kt
package kr.co.ongil.data.model.user

import kotlinx.serialization.Serializable

/**
 * 사용자 정보 조회 API 응답
 * GET /api/v1/users/me
 *
 * 실제 서버 응답:
 * {
 *   "message": "...",
 *   "data": {
 *     "id": 1,
 *     "name": "홍길동",
 *     "birth": "19980919",
 *     "phoneNumber": "01012341234",
 *     "userType": "PATIENT",
 *     "profileImage": "S3_URL"
 *   }
 * }
 */
@Serializable
data class UserResponse(
    val message: String,
    val data: UserDto  // data 바로 아래에 유저 정보
)
